runtime/context.go

```diff
@@ -244,19 +259,20 @@ func (c *ContextExtractors) ExtractLogFields(ctx context.Context) []zap.Field {
 }
 
 // ContextLogger is a logger that extracts some log fields from the context before passing through to underlying zap logger.
+//In cases it also updates the context instead of logging
 type ContextLogger interface {
-	Debug(ctx context.Context, msg string, fields ...zap.Field)
-	Error(ctx context.Context, msg string, fields ...zap.Field)
-	Info(ctx context.Context, msg string, fields ...zap.Field)
-	Panic(ctx context.Context, msg string, fields ...zap.Field)
-	Warn(ctx context.Context, msg string, fields ...zap.Field)
-
-	// DebugZ skips logs if skipZanzibarLogs is set to true otherwise behaves as normal Debug, similarly for other XxxxZ's
-	DebugZ(ctx context.Context, msg string, fields ...zap.Field)
-	ErrorZ(ctx context.Context, msg string, fields ...zap.Field)
-	InfoZ(ctx context.Context, msg string, fields ...zap.Field)
-	PanicZ(ctx context.Context, msg string, fields ...zap.Field)
-	WarnZ(ctx context.Context, msg string, fields ...zap.Field)
+	Debug(ctx context.Context, msg string, fields ...zap.Field) context.Context
+	Error(ctx context.Context, msg string, fields ...zap.Field) context.Context
+	Info(ctx context.Context, msg string, fields ...zap.Field) context.Context
+	Panic(ctx context.Context, msg string, fields ...zap.Field) context.Context
+	Warn(ctx context.Context, msg string, fields ...zap.Field) context.Context
+
+	// DebugZ skips logs, and adds to context if skipZanzibarLogs is set to true otherwise behaves as normal Debug, similarly for other XxxxZ's
+	DebugZ(ctx context.Context, msg string, fields ...zap.Field) context.Context
```

Comment:
What is the purpose of returning the context ? If we pass the context pointer then returning context might not be required.

Reply:
`context.Context` interface has fields that point to pointers under the hood only. 
We need not pass a pointer to a context.

Reply:
The standard way of using the context is to update the context and return the context. Do we see any reason why we should pass a pointer and not return a context?